apache/fineract-cn-mobile
# Added UI/Integration tests for Create/edit, view and list products

This PR has been made on top of this [PR](https://github.com/apache/fineract-cn-mobile/pull/192)

Please Add Screenshots If any UI changes.

1. Create Product using couchbase successfully 
- Enter Product identifier
- Enter name 
- Enter pattern package 
- Enter the temporal unit 
- Enter maximum Term Range 
- Enter min. balance range 
- Enter max. balance range 
- Enter min. Interest Range 
- Enter max. Interest Range
- Navigate to the next step 
- Add an account assignment 
- Enter designator, account identifier, ledger identifier 
- Edit designator field of account assignment 
- test delete account assignment 
- Assert if the values in the product review step are correctly populated 
- Check if the product was created successfully by querying sync-gateway for the desired product 
- delete the test product entry from sync-gateway replicator 


https://user-images.githubusercontent.com/53621853/130319308-0bbff424-d8d6-4579-aa76-bf3f8c733360.mp4


2. Test product list fragment 
- Launch the application 
- Navigate to the product list fragment and check if the recyclerView is visible 
- Click on the recyclerView utem 
- Assert if the product details screen shows the correct data 



- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything.

- [x] If you have multiple commits please combine them into one commit by squashing them.
